keps/sig-api-machinery/20190226-network-proxy.md

```diff
@@ -241,102 +241,29 @@ or which user/tenant tries to initiate the request, etc.
 
 ### Proxy gRPC definition
 
+In order to serve a proxy request, one gRPC bidirectional stream on proxy
+server is created to serve it. It's a 1:1 mapping from TCP connection to a
+gRPC stream, so the state of TCP connection is exactly the same as the gRPC
+stream state.
+
 ```grpc
 syntax = "proto3";
 
 service ProxyService {
-    // Proxy connects to a remote address by stream id, and establish
-    // a bi-directional stream of packet exchange.
-    rpc Proxy(stream Packet) returns (stream Packet) {}
-}
-
-enum PacketType {
-  DIAL_REQ = 0;
-  DIAL_RSP = 1;
-  CLOSE_REQ = 2;
-  CLOSE_RSP = 3;
-  DATA = 4;
-}
-
-message Packet {
-  PacketType type = 1;
-
-  oneof payload {
-    DialRequest dialRequest = 2;
-    DialResponse dialResponse = 3;
-    CloseRequest closeRequest = 4;
-    CloseResponse closeResponse = 5;
-    Data data = 6;
-  }
-}
-
-// Error is sent when error happens from remote side of connection when it tries
-// to read or write from it
-enum Error {
-  EOF = 0;
-  EIO = 1;
-  ECONNRESET = 2;
-  ETIMEOUT = 3;
-  EADDRNOTAVAIL = 4;
-  EMFILE = 5;
-  // … 
+  // Proxy a TCP connection to a remote address defined by ConnectParam.
+  // The ConnectParam is defined in metadata under key "x-kube-proxy/v1".
+  // metadata["x-kube-proxy/v1"] = base64.Encode(proto.Marshal(connectParam))
```

Comment:
Couple things:
1) base64 is not necessary over grpc or http2
2) "kube-proxy" also means something very different than what you are doing here
3) Why would version the header vs versioning the entire service?

Reply:
2-3) +1. version will cause confusing. I will remove it. 
1) I want the metadata value is really a valid string.